core/domain/user_services.py

```diff
@@ -551,14 +942,46 @@ def get_email_preferences(user_id):
 
 
 class UserContributions(object):
-    """Value object representing a user's contributions."""
+    """Value object representing a user's contributions.
+
+    Attributes:
+        user_id: str. The user id.
+        created_exploration_ids: list. IDs of explorations that this
+            user has created.
+        edited_exploration_ids: list. IDs of explorations that this
+        	user has edited.
+    """
+
     def __init__(
             self, user_id, created_exploration_ids, edited_exploration_ids):
+        """Constructs a UserContributions domain object.
+
+        Args:
+            user_id: str. The user id.
+            created_exploration_ids: list. IDs of explorations that this
+                user has created.
+            edited_exploration_ids: list. IDs of explorations that this
+                user has edited.
+        """
         self.user_id = user_id
         self.created_exploration_ids = created_exploration_ids
         self.edited_exploration_ids = edited_exploration_ids
 
     def validate(self):
+        """Checks that user_id, created_exploration_ids and
+        edited_exploration_ids fields of this UserContributions
+        domain object are valid.
+
+        Raises:
+            ValidationError: If user_id is not str.
```

Comment:
Drop the "if".

ValidationError: user_id is not a str.
ValidationError: created_exploration_ids is not a list.

etc.